spec/safe_statements_spec.rb

```diff
@@ -2365,6 +2365,45 @@ def up
             test_migration.suppress_messages { test_migration.migrate(:up) }
           end
 
+          it "create an index with nulls_not_distinct option" do
+            create_range_partitioned_table(:foos3, migration_klass, with_partman: true)
+
+            test_migration = Class.new(migration_klass) do
+              def up
+                safe_add_concurrent_partitioned_index :foos3, :updated_at, nulls_not_distinct: true
+              end
+            end
+
+            allow(ActiveRecord::Base.connection).to receive(:execute).and_call_original
+
+            aggregate_failures do
+              expect(ActiveRecord::Base.connection).to receive(:execute).with(/CREATE INDEX "index_foos3_on_updated_at" ON ONLY/).once.ordered
+              expect(ActiveRecord::Base.connection).to receive(:execute).with(/CREATE INDEX CONCURRENTLY/).exactly(10).times.ordered
+              expect(ActiveRecord::Base.connection).to receive(:execute).with(/ALTER INDEX .+\nATTACH PARTITION/).exactly(10).times.ordered
+            end
```

Comment:
This is exactly what I would have done. The expectations are way more precise and the style matches other nearby tests

_However_, I think we should also have expectations that the SQL includes `NULLS NOT DISTINCT`